hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/schema/MappingCatalog.java

```diff
@@ -133,4 +141,24 @@ private Table toTable(Mapping mapping) {
                 mapping.fields()
         );
     }
+
+    @Override
+    public void registerListener(TableListener listener) {
+        listeners.add(listener);
+        storage.registerListener(new MappingStorage.EntryListenerAdapter() {
+            @Override
+            public void entryUpdated(EntryEvent<String, Mapping> event) {
+                if (!event.getMember().localMember()) {
```

Comment:
Why don't we handle also local members here, we won't need the special code above in `createMapping()` and `removeMapping()`

Reply:
The thing is that listeners are executed asynchronously, in that case plans would not be guaranteed to be invalidated even on local node after we respond to the client.

Reply:
True, makes sense. But it's worth adding a comment about that.